### PR TITLE
Added sender email

### PR DESCRIPTION
**Pull request by DanilBut**

### CODE REVIEW

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/EmailConfig.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @Value("${spring.mail.port}")
+    private int port;
+    @Value("${spring.mail.username}")
+    private String username;
+    @Value("${spring.mail.password}")
+    private String password;
+
+    @Bean
+    public JavaMailSender javaMailSender() {
+        JavaMailSenderImpl mailSender = new JavaMailSenderImpl();
```
Comment (akarzhaneuski) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/EmailConfig.java:27]: ``` JavaMailSender mailSender = new JavaMailSenderImpl(); ```
see https://habr.com/ru/post/83269/ as a reference

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/CandidateServiceImpl.java [hunk 2]
```diff
@@ -62,7 +71,19 @@ public CandidateResponseDTO findById(Long id) {
...
+
+        MessageDTO messageDTO = new MessageDTO();
+        messageDTO.setTo(candidateCreateDTO.getEmail());
+        messageDTO.setSubject(EmailConstant.SUBJECT_RECEIVED_RESUME);
+        messageDTO.setProperties(prop);
+        messageDTO.setTemplateName("receivingCandidateResume.ftl");
+        try {
+            sendEmailService.sendEmail(messageDTO);
+        } catch (TemplateException | IOException exception) {
+            exception.printStackTrace();
```
Comment (akarzhaneuski) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/CandidateServiceImpl.java:85]: Bad exception handling, its better to log exception with message like:
``` log.errror("Error during email sending {}", e.getMessage(), e); ```

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/InterviewTimeServiceImpl.java [hunk 3]
```diff
@@ -58,8 +70,23 @@ public InterviewTimeResponseDTO findById(Long id) {
...
+
+        MessageDTO messageDTO = new MessageDTO();
+        messageDTO.setTo(candidateResponseDTO.getEmail());
+        messageDTO.setSubject(EmailConstant.SUBJECT_INTERVIEW_TIME);
+        messageDTO.setProperties(prop);
+        messageDTO.setTemplateName("interviewTimeEmail.ftl");
+        try {
+            sendEmailService.sendEmail(messageDTO);
+        } catch (IOException | TemplateException e) {
+            e.printStackTrace();
```
Comment (akarzhaneuski) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/InterviewTimeServiceImpl.java:86]: same about logs

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/SendEmailServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        MimeMessage mimeMessage = javaMailSender.createMimeMessage();
+        MimeMessageHelper mimeMessageHelper = null;
+        try {
+            mimeMessageHelper = new MimeMessageHelper(mimeMessage, true, "UTF-8");
+            mimeMessageHelper.setFrom(messageDTO.getFrom());
+            mimeMessageHelper.setTo(messageDTO.getTo());
+            mimeMessageHelper.setSubject(messageDTO.getSubject());
+            mimeMessageHelper.setText(messageDTO.getText(), true);
+        } catch (MessagingException e) {
+            e.printStackTrace();
```
Comment (akarzhaneuski) [inline sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/SendEmailServiceImpl.java:53]: same about logs

**Review by akarzhaneuski — CHANGES_REQUESTED** [review]

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/EmailConfig.java (hunk 1 above)
Comment (DanilBut) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/configuration/EmailConfig.java]: i have moved the config to application.properties

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/CandidateServiceImpl.java (hunk 2 above)
Comment (DanilBut) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/CandidateServiceImpl.java]: corrected

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/InterviewTimeServiceImpl.java (hunk 3 above)
Comment (DanilBut) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/InterviewTimeServiceImpl.java]: corrected

File: sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/SendEmailServiceImpl.java (hunk 4 above)
Comment (DanilBut) [reply sandbox-backend/src/main/java/com/exadel/team2/sandbox/service/impl/SendEmailServiceImpl.java]: corrected

**Review by akarzhaneuski — APPROVED** [review]

**Pull request merged**